check_pip_requirements.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+set -ex
+
+reqs=$1
+pinned=$2
+new_pinned=$(mktemp)
+pinned_no_comments=$(mktemp)
+new_pinned_no_comments=$(mktemp)
+
+pip-compile --quiet $reqs $pinned --output-file=$new_pinned
+# Get rid of comments that might differ despite requirements being the same
+cat $pinned | sed '/#/d' > $pinned_no_comments
+cat $new_pinned | sed '/#/d' > $new_pinned_no_comments
+diff $pinned_no_comments $new_pinned_no_comments
```

Comment:
Do you need to exit with the exit code from `diff`?

Reply:
The script should inherit the exit code of its last command, and because we have `set -e`, then it should fail if diff comes up different. Here's an example locally after I added an unrelated dependency, `msprime` to the hail/python/requirements.txt:

```
./check_pip_requirements.sh hail/python/requirements.txt hail/python/pinned-requirements.txt
+ reqs=hail/python/requirements.txt
+ pinned=hail/python/pinned-requirements.txt
++ mktemp
+ new_pinned=/var/folders/2c/0b40ckvn7bq_g0zz9qtkxwr40000gp/T/tmp.FxFFOFRJ
++ mktemp
+ pinned_no_comments=/var/folders/2c/0b40ckvn7bq_g0zz9qtkxwr40000gp/T/tmp.70s9RDbL
++ mktemp
+ new_pinned_no_comments=/var/folders/2c/0b40ckvn7bq_g0zz9qtkxwr40000gp/T/tmp.IjRW9sSq
+ pip-compile --quiet hail/python/requirements.txt hail/python/pinned-requirements.txt --output-file=/var/folders/2c/0b40ckvn7bq_g0zz9qtkxwr40000gp/T/tmp.FxFFOFRJ
+ cat hail/python/pinned-requirements.txt
+ sed /#/d
+ cat /var/folders/2c/0b40ckvn7bq_g0zz9qtkxwr40000gp/T/tmp.FxFFOFRJ
+ sed /#/d
+ diff /var/folders/2c/0b40ckvn7bq_g0zz9qtkxwr40000gp/T/tmp.70s9RDbL /var/folders/2c/0b40ckvn7bq_g0zz9qtkxwr40000gp/T/tmp.IjRW9sSq
21a22
> demes==0.2.2
33a35,36
> importlib-metadata==4.11.4
> importlib-resources==5.7.1
37a41
> jsonschema==4.6.0
40a45
> msprime==1.2.0
43a49
> newick==1.3.2
59a66
> pyrsistent==0.18.1
67a75,76
> ruamel-yaml==0.17.21
> ruamel-yaml-clib==0.2.6
71a81
> svgwrite==1.4.2
75a86
> tskit==0.4.1
80a92
> zipp==3.8.0
make: *** [check-pip-dependencies] Error 1
```